jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Stream.java

```diff
@@ -413,6 +416,8 @@ private void onData(DataFrame frame, Callback callback)
         DataEntry entry = new DataEntry(frame, callback);
         try (AutoLock l = lock.lock())
         {
+            if (dataQueue == null)
+                dataQueue = new ArrayDeque<>(8);
```

Comment:
Why specifying the element count here? It wasn't when the `dataQueue` was initialized at the field level.

Reply:
Can you set it as a function of the actual flow control window?